dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/operator/BaseTaskExecuteRunnableTimeoutOperator.java

```diff
@@ -38,18 +38,8 @@ public BaseTaskExecuteRunnableTimeoutOperator(TaskInstanceDao taskInstanceDao) {
 
     @Override
     public void operate(DefaultTaskExecuteRunnable taskExecuteRunnable) {
-        // Right now, if the task is running in worker, the timeout strategy will be handled at worker side.
-        // if the task is in master, the timeout strategy will be handled at master side.
-        // todo: we should unify this, the master only need to handle the timeout strategy. and send request to worker
-        // to kill the task, if the strategy is timeout_failed.
         TaskInstance taskInstance = taskExecuteRunnable.getTaskInstance();
         TaskTimeoutStrategy taskTimeoutStrategy = taskInstance.getTaskDefine().getTimeoutNotifyStrategy();
-        if (TaskTimeoutStrategy.FAILED != taskTimeoutStrategy
```

Comment:
if `taskTimeoutStrategy ` == `TaskTimeoutStrategy.FAILED`, it would not step into this block, I don't get how this PR could fix this problem, could you please add some description?

Reply:
based on my research, 
in originnal case, it will directly return when  `taskTimeoutStrategy ==TaskTimeoutStrategy.FAILED`,
if we remove this statment, it will trigger `killRemoteTaskInstanceInThreadPool(taskInstance)` and will finally trigger `task.cancel()`

Reply:
I'm a little confused, I think it will also trigger `killRemoteTaskInstanceInThreadPool(taskInstance)` in line 45 when taskTimeoutStrategy  == TaskTimeoutStrategy.FAILED

Reply:
Thanks @rickchengx 

I am afraid that  you are right.
https://github.com/apache/dolphinscheduler/blob/dev/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/event/TaskTimeoutStateEventHandler.java#L61-L65

Hi, @ruanwenjun ,
I saw current code will kill all ongoing task if failurestrategy is set to END.
https://github.com/apache/dolphinscheduler/blob/dev/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java#L410C1-L412C22

Will you like to provide more context about this issue?
I will repro it in my local to see what it is the root cause is.

Reply:
Yes, right now, we didn't have this problem, I will close the issue.